src/protagonist/Engine/Data/EngineAssetRepository.cs

```diff
@@ -85,6 +85,15 @@ public EngineAssetRepository(DlcsContext dlcsContext, ILogger<EngineAssetReposit
     public ValueTask<Asset?> GetAsset(AssetId assetId, CancellationToken cancellationToken = default)
         => dlcsContext.Images.FindAsync(new object[] { assetId }, cancellationToken);
 
+    public async Task<long?> GetImageSize(AssetId assetId, CancellationToken cancellationToken = default)
+    {
+        var imageSize = await dlcsContext.ImageStorages.AsNoTracking().FirstOrDefaultAsync(

```

Comment:
As only `Size` is used there's no need to select the entire object (not a huge perf saving in this instance as it's a small table).

```cs
var imageSize = await dlcsContext.ImageStorages
  .Where(i => i.Id == assetId)
  .Select(i => i.Size)
  .FirstOrDefaultAsync();
```